src/index.js

```diff
@@ -1,390 +1,586 @@
-export function h(name, attributes) {
-  var rest = []
-  var children = []
-  var length = arguments.length
+var DEFAULT = 0
+var RECYCLED_NODE = 1
+var TEXT_NODE = 2
 
-  while (length-- > 2) rest.push(arguments[length])
+var XLINK_NS = "http://www.w3.org/1999/xlink"
+var SVG_NS = "http://www.w3.org/2000/svg"
 
-  while (rest.length) {
-    var node = rest.pop()
-    if (node && node.pop) {
-      for (length = node.length; length--; ) {
-        rest.push(node[length])
+var EMPTY_OBJECT = {}
+var EMPTY_ARRAY = []
```

Comment:
:thinking:

Reply:
A small optimization, mainly used for text vnodes that don't have children, but still use the same VNode shape.